src/simplewallet/simplewallet.cpp

```diff
@@ -3381,6 +3451,129 @@ bool simple_wallet::check_tx_key(const std::vector<std::string> &args_)
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::check_tx_proof(const std::vector<std::string> &args)
+{
+  if(args.size() != 3) {
+    fail_msg_writer() << tr("usage: check_tx_proof <txid> <address> <signature>");
+    return true;
+  }
+
+  if (!try_connect_to_daemon())
+    return true;
+
+  // parse txid
+  assert(m_wallet);
```

Comment:
Again, this exists because I copied the code from `check_tx_key()`, and I see 5 other occurrences of it in simplewallet.cpp. First, how should it be changed? Should it print an error message by `fail_msg_writer()` and return true, or should it throw like `throw std::runtime_error("simple_wallet null wallet");` as done in `get_daemon_blockchain_height()`? Second, should I also change other parts in simplewallet.cpp doing the same?